### ISSUE TITLE
Regression on include file in v1.3

### ISSUE THREAD

**Issue by fenxiong** [fluent/fluent-bit#1674]
## Bug Report

**Describe the bug**

Before fluentbit 1.3, the config file specified in `@INCLUDE` section does not need to contain a newline at the end. Starting from fluentbit 1.3, not having a newline at the end of the included file resulted in fluentbit failed to start:
```
Fluent Bit v1.3.0
Copyright (C) Treasure Data

[2019/10/21 22:29:12] [  Error] File /fluent-bit/etc/external.conf
[2019/10/21 22:29:12] [  Error] Error in line 6: Length of content has execeded limit
```

**To Reproduce**
- Rubular link if applicable:
- Example log message if applicable:
```
{"log":"YOUR LOG MESSAGE HERE","stream":"stdout","time":"2018-06-11T14:37:30.681701731Z"}
```
- Steps to reproduce the problem:
The below steps show the config files i use, and that the same config files work for fluentbit 1.2.2, but not for fluentbit 1.3.0.
```
[ec2-user@ip-172-31-22-2 test]$ pwd
/home/ec2-user/test
[ec2-user@ip-172-31-22-2 test]$ ls
external.conf  fluent-bit.conf
[ec2-user@ip-172-31-22-2 test]$ cat external.conf
[INPUT]
    Name mem
    Tag  mem.local

[OUTPUT]
    Name stdout
    Match mem.local[ec2-user@ip-172-31-22-2 test]$
[ec2-user@ip-172-31-22-2 test]$ cat fluent-bit.conf
[INPUT]
    Name forward
    Listen 0.0.0.0
    Port 24224

@INCLUDE /fluent-bit/etc/external.conf

[OUTPUT]
    Name null[ec2-user@ip-172-31-22-2 test]$
[ec2-user@ip-172-31-22-2 test]$ docker run -v /home/ec2-user/test/external.conf:/fluent-bit/etc/external.conf -v /home/ec2-user/test/fluent-bit.conf:/fluent-bit/etc/fluent-bit.conf -d fluent/fluent-bit:1.2.2
e42cf902a279a0da6049779d7b9099421bc1614a81759bf88619c73c952aed61
[ec2-user@ip-172-31-22-2 test]$ docker ps
CONTAINER ID        IMAGE                     COMMAND                  CREATED             STATUS              PORTS               NAMES
e42cf902a279        fluent/fluent-bit:1.2.2   "/fluent-bit/bin/flu…"   3 seconds ago       Up 2 seconds        2020/tcp            distracted_edison
[ec2-user@ip-172-31-22-2 test]$ docker logs e42cf902a279
Fluent Bit v1.2.2
Copyright (C) Treasure Data

[2019/10/21 22:28:58] [ info] [storage] initializing...
[2019/10/21 22:28:58] [ info] [storage] in-memory
[2019/10/21 22:28:58] [ info] [storage] normal synchronization mode, checksum disabled, max_chunks_up=128
[2019/10/21 22:28:58] [ info] [engine] started (pid=1)
[2019/10/21 22:28:58] [ info] [in_fw] binding 0.0.0.0:24224
[2019/10/21 22:28:58] [ warn] [router] NO match for null.1 output instance
[2019/10/21 22:28:58] [ warn] [router] NO match for null.1 output instance
[2019/10/21 22:28:58] [ info] [sp] stream processor started
[0] mem.local: [1571696939.000155839, {"Mem.total"=>16424240, "Mem.used"=>4631604, "Mem.free"=>11792636, "Swap.total"=>0, "Swap.used"=>0, "Swap.free"=>0}]
[1] mem.local: [1571696940.000115685, {"Mem.total"=>16424240, "Mem.used"=>4631568, "Mem.free"=>11792672, "Swap.total"=>0, "Swap.used"=>0, "Swap.free"=>0}]
[2] mem.local: [1571696941.000088148, {"Mem.total"=>16424240, "Mem.used"=>4636536, "Mem.free"=>11787704, "Swap.total"=>0, "Swap.used"=>0, "Swap.free"=>0}]
[3] mem.local: [1571696942.000108579, {"Mem.total"=>16424240, "Mem.used"=>4629536, "Mem.free"=>11794704, "Swap.total"=>0, "Swap.used"=>0, "Swap.free"=>0}]
[ec2-user@ip-172-31-22-2 test]$ docker run -v /home/ec2-user/test/external.conf:/fluent-bit/etc/external.conf -v /home/ec2-user/test/fluent-bit.conf:/fluent-bit/etc/fluent-bit.conf -d fluent/fluent-bit:1.3.0
94697205b707332109ef9b595f9be387baebfe89f813a6c388adbe88b363bf4b
[ec2-user@ip-172-31-22-2 test]$ docker ps -a
CONTAINER ID        IMAGE                     COMMAND                  CREATED             STATUS                     PORTS               NAMES
94697205b707        fluent/fluent-bit:1.3.0   "/fluent-bit/bin/flu…"   2 seconds ago       Exited (1) 2 seconds ago                       optimistic_wiles
e42cf902a279        fluent/fluent-bit:1.2.2   "/fluent-bit/bin/flu…"   16 seconds ago      Up 16 seconds              2020/tcp            distracted_edison
[ec2-user@ip-172-31-22-2 test]$ docker logs 94697205b707
Fluent Bit v1.3.0
Copyright (C) Treasure Data

[2019/10/21 22:29:12] [  Error] File /fluent-bit/etc/external.conf
[2019/10/21 22:29:12] [  Error] Error in line 6: Length of content has execeded limit
```

**Expected behavior**

Include file behavior works the same for newer versions - don't need a newline at the end, like before.

**Screenshots**

**Your Environment**

* Version used: 1.3.0
* Configuration: shown above
* Environment name and version (e.g. Kubernetes? What version?): 1.3.0 fluentbit image from Dockerhub
* Server type and version: 1.3.0
* Operating System and version: Amazon Linux 2 AMI
* Filters and plugins: built-in plugin

**Additional context**

I did not find anything related to newline in https://fluentbit.io/announcements/v1.3.0/, so i assume this behavior change isn't expected.

**Comment by PettitWesley** [comment]
I'm seeing the same thing with `1.3.2`

**Comment by edsiper** [comment]
Your configuration file is missing a final ```\n``` . Previous versions did not warn about this and lead to undefined behaviors since last option in the file is not complete.

As a pending improvement, we will make sure the configuration reader reads the config value if it reaches EOF.

**Comment by rmacian** [comment]
This issue should be labeled as bug, not enhacement. Editing a configmap with a conf.file from openshift web ui breaks the config and has to be manually edited

**Comment by edsiper** [comment]
Yeah, you are correct. Already re-labeled as a bug.

I've pushed a fix into GIT master for this problem, the fix will be part of the next  v1.3 release next week (53dcfc39)
